src/schema/v2/createPartnerOfferMutation.ts

```diff
@@ -57,22 +58,32 @@ export const createPartnerOfferMutation = mutationWithClientMutationId<
     discount_percentage: { type: GraphQLInt },
   },
   outputFields: {
+    partner: {
+      type: PartnerType,
+      resolve: (result) => result.partner || null,
+    },
     partnerOfferOrError: {
       type: ResponseOrErrorType,
       description: "On success: the partner offer created.",
-      resolve: (result) => result,
+      resolve: (result) => result.partnerOffer || result,
     },
   },
-  mutateAndGetPayload: async (args, { createPartnerOfferLoader }) => {
+  mutateAndGetPayload: async (
+    args,
+    { createPartnerOfferLoader, partnerAllLoader }
+  ) => {
     if (!createPartnerOfferLoader) {
       throw new Error("You need to be signed in to perform this action")
     }
 
     try {
-      return await createPartnerOfferLoader?.({
+      const partnerOffer = await createPartnerOfferLoader?.({
         artwork_id: args.artwork_id,
         discount_percentage: args.discount_percentage,
       })
+      const partner = await partnerAllLoader?.(partnerOffer.partner_id)
```

Comment:
don't fetch the `partner` pre-emptively, move this to the `partner` resolver in the payload, only if a client requests it.